### ISSUE TITLE
PostgreSQL plugin creates roles with password in clear text

### ISSUE THREAD

**Issue by rafaelremondes** [hashicorp/vault#6910]
Hello

I am using the PostgreSQL plugin to create temporary credentials. The creation statement allows to set a password for the role newly created. However, in PostgreSQL, when creating a ROLE with password it is showed in clear test in logs.

Documentation in Postgres suggests that is possible to pass an hash using md5 or other function instead of clear text password. 

https://www.postgresql.org/docs/current/sql-createrole.html

```
If the presented password string is already in MD5-encrypted or SCRAM-encrypted format, then it is stored as-is regardless of password_encryption (since the system cannot decrypt the specified encrypted password string, to encrypt it in a different format). This allows reloading of encrypted passwords during dump/restore.
```
So, could Vault store an hashed password instead of one in clear text?

Disabling the logs for the db user Vault uses to create the role would work but it would not be ideal as we would miss logs that might be important for debugging or auditing.

Thanks

**Comment by evkuzin** [comment]
agreed, I have the same problem. When logs format == ddl - all passwords stored cleartext in log.

**Comment by ncabatoff** [comment]
Think we would need https://github.com/lib/pq/issues/941 (or an equivalent from another library) to implement this.

**Comment by Neustradamus** [comment]
@rafaelremondes, @evkuzin, @ncabatoff: The lib/pq has SCRAM-SHA-256 support since a moment:
- https://github.com/lib/pq/pull/833

And you can see this PR here:
- https://github.com/hashicorp/vault/pull/6895

**Comment by aphorise** [comment]
There's been a change to **pgx** from ~lib/pq~ as per the [CHANGELOG.md release notes of 1.11.0](https://github.com/hashicorp/vault/blob/main/CHANGELOG.md#1110)

> database & storage: Change underlying driver library from [lib/pq](https://github.com/lib/pq) to [pgx](https://github.com/jackc/pgx). This change affects Redshift & Postgres database secrets engines, and CockroachDB & Postgres storage engines [GH-15343](https://github.com/hashicorp/vault/pull/15343).

I'm curious if this is this an issue or relevant?

@rafaelremondes can you kindly confirm if you've rested since?

PS - @Neustradamus @evkuzin - any input from you folks on a retest would be welcome too.

**Comment by raymonstah** [comment]
Hey folks, there's a fix for this in https://github.com/hashicorp/vault/pull/19616 and should be released as a part of Vault 1.14.
You'll have to update your PostgreSQL configuration using the following:

```
    $ vault write database/config/my-postgresql-database \
        ... \
        password_authentication="scram-sha-256"
```
With the password_authentication set to scram-sha-256, passwords created by Vault will first hashed before sending them to PostgreSQL, which will store the hashed password as-is, preventing plaintext passwords from leaking in the logs.
